packages/x-charts/package.json

```diff
@@ -71,12 +71,14 @@
     ".": {
       "types": "./index.d.ts",
       "require": "./index.js",
-      "import": "./esm/index.js"
+      "import": "./esm/index.js",
+      "default": "./modern/index.js"
```

Comment:
@LukasTy Is this still necessary? I'm hesitant adding more conditions if they don't solve an actual issue.

Reply:
Yes, it is the main change that fixed all the demos.
Without it none of the demos load in codesandbox.
The changes on one of the demos is for import depth fixing. 😉

Reply:
It's probably something that `codesandbox` is doing on their side, because StackBlitz is working fine. 🤷

Reply:
Yep, it feels a bit like we're solving the problem in the wrong location. When I fix the codesandbox so that the files are actually covered by the `tsconfig.json` I get a different error: https://codesandbox.io/s/peaceful-austin-8ktz7l?file=/tsconfig.json

Reply:
Or we could focus on solving https://github.com/mui/mui-x/issues/10257?
Maybe solving it could also cover the issue that codesandbox is having. 🤔

Reply:
It seems that just removing the `require` condition also fixes the issue: https://github.com/mui/mui-x/pull/10272
It's a problematic condition because it wouldn't work in the first place. That code is trying to `require` a ESM only module (`d3`). 
Still no clue as to why `react-scripts` in this specific codesandbox setup gives precedence of the `default` condition though.

Reply:
Feel free to incorporate https://github.com/mui/mui-x/pull/10272 in this PR and merge.

Reply:
Or maybe just merge your PR specifically targeting the fix you've done, invested time, and resolved? 🤔